runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -1799,6 +1809,33 @@ TR_RelocationRecordThunks::relocateAndRegisterThunk(TR_RelocationRuntime *reloRu
    return 0;
    }
 
+// TR_J2IVirtualThunkPointer Relocation
+char *
+TR_RelocationRecordJ2IVirtualThunkPointer::name()
+   {
+   return "TR_J2IVirtualThunkPointer";
+   }
+
+int32_t
+TR_RelocationRecordJ2IVirtualThunkPointer::bytesInHeaderAndPayload()
+   {
+   return sizeof(TR_RelocationRecordJ2IVirtualThunkPointerBinaryTemplate);
+   }
+
+void
+TR_RelocationRecordJ2IVirtualThunkPointer::relocateJ2IVirtualThunkPointer(
+   TR_RelocationTarget *reloTarget,
+   uint8_t *reloLocation,
+   void *thunk)
+   {
+   // For uniformity with TR_Thunks, the reloLocation is not the location of the
+   // J2I thunk pointer, but rather the location of the constant pool address.
+   // Find the J2I thunk pointer relative to that.
+   auto recordData = (TR_RelocationRecordJ2IVirtualThunkPointerBinaryTemplate *)_record;
+   reloLocation += recordData->_offsetToJ2IVirtualThunkPointer;
```

Comment:
I think you're missing the `preparePrivateData()` code here....nothing sets _offsetToJ2IVirtualThunkPointer from the binary relocation data, does it? Or did I miss it?

Reply:
Doesn't `_record` point into the binary relocation data? I do get the correct offset (when I have run with the further change in #2875)

Reply:
Updated to define and call a helper method that uses `reloTarget` to read `_offsetToJ2IVirtualThunkPointer`